packages/client-react-streaming/src/SimulatePreloadedQuery.cc.ts

```diff
@@ -1,80 +1,33 @@
 "use client";
 
+import { useApolloClient, useBackgroundQuery } from "@apollo/client/index.js";
+import { useMemo, type ReactNode } from "react";
 import {
-  skipToken,
-  useApolloClient,
-  useBackgroundQuery,
-} from "@apollo/client/index.js";
-import type { ApolloClient as WrappedApolloClient } from "./DataTransportAbstraction/WrappedApolloClient.js";
-import type {
-  ProgressEvent,
-  TransportIdentifier,
-} from "./DataTransportAbstraction/DataTransportAbstraction.js";
-import {
-  deserializeOptions,
-  type TransportedOptions,
-} from "./DataTransportAbstraction/transportedOptions.js";
-import type { QueryManager } from "@apollo/client/core/QueryManager.js";
-import { useMemo } from "react";
-import type { ReactNode } from "react";
-import invariant from "ts-invariant";
-import type { TransportedQueryRefOptions } from "./transportedQueryRef.js";
+  reviveTransportedQueryRef,
+  type TransportedQueryRef,
+} from "./transportedQueryRef.js";
+import { deserializeOptions } from "./DataTransportAbstraction/transportedOptions.js";
 import type { PreloadQueryOptions } from "./PreloadQuery.js";
 
-const handledRequests = new WeakMap<TransportedOptions, TransportIdentifier>();
-
 export default function SimulatePreloadedQuery<T>({
-  options,
-  result,
+  queryRef,
   children,
-  queryKey,
 }: {
-  options: TransportedQueryRefOptions;
-  result: Promise<Array<Omit<ProgressEvent, "id">>>;
+  queryRef: TransportedQueryRef<T>;
   children: ReactNode;
-  queryKey?: string;
 }) {
-  const client = useApolloClient() as WrappedApolloClient<any>;
-  if (!handledRequests.has(options)) {
-    const id =
-      `preloadedQuery:${(client["queryManager"] as QueryManager<any>).generateQueryId()}` as TransportIdentifier;
-    handledRequests.set(options, id);
-    invariant.debug(
-      "Preloaded query %s started on the server, simulating ongoing request",
-      id
-    );
-    client.onQueryStarted!({
-      type: "started",
-      id,
-      options,
-    });
-
-    result.then((results) => {
-      invariant.debug("Preloaded query %s: received events: %o", id, results);
-      for (const event of results) {
-        client.onQueryProgress!({ ...event, id } as ProgressEvent);
-      }
-    });
-  }
+  const client = useApolloClient();
+  reviveTransportedQueryRef(queryRef, client);
```

Comment:
All of that logic now lives in `reviveTransportedQueryRef` as it will also be used by other framework implementations.